### PR TITLE
feature (common) added [FRONT] Funcionalidad Dinámica de Búsqueda [SEM-7]

### PR DESCRIPTION
**Pull request by Timois**
…M - 7]

### CODE REVIEW

**Comment by olmedo90** [comment]
image
It is not according to the figma. add appropriate styles for the search section.
Also fix component overflow.
👇🏼👇🏼👇🏼
image

File: src/components/Navbar/index.tsx [hunk 1]
```diff
@@ -1,20 +1,124 @@
...
+import SearchIcon from '../../assetss/searchIcon.svg'
+
+// Para usar la api primero se debe comentar el codigo que esta moqueando los datos, en este caso seria las lineas
+// 9, 15 y 16 y descomentar las lineas 17 y 18 junto con la funcion de useEffect de la linea 61 al 70.
+import dataJson from '../__mock__/data.json'
+//const endpoint = "https://fakestoreapi.com/products" La url de la api tiene que ser pegado aqui.
 const Navbar = () => {
+    const [toogle, setToogle] = useState(false)
+
+    const [formData, setFormData] = useState('')
```
Comment (Ditmar) [inline src/components/Navbar/index.tsx:14]: ok, all this code can be changed using another approach let me explain to you.
When you are coding try to think of the best way to write the code following the DRY principles, in this case, you need to take the data from the forms. in this case, you can create a new hook for example called useForm.ts
```
import { useState, ChangeEvent } from 'react';

type FormState = { [key: string]: any };

const useForm = (initForm: FormState) => {
    const [form, setForm] = useState<FormState>(initForm);

    const handlerChangeForm = ({ target }: ChangeEvent<HTMLInputElement>) => {
        setForm({...form, [target.name]: target.value});
    }

    const handlerResetForm = () => {
        setForm(initForm);
    }

    return [form, handlerChangeForm, handlerResetForm];
}

export default useForm;
```
And you can use it.

```
import React, { FC } from 'react';
import useForm from './useForm'; // Asume que useForm está en el mismo directorio

const UserForm: FC = () => {
    const initialFormState = { name: '', email: '' };
    const [form, handlerChangeForm, handlerResetForm] = useForm(initialFormState);

    const handleSubmit = (event: React.FormEvent) => {
        event.preventDefault();
        console.log('Form Submitted:', form);
        handlerResetForm();
    }

    return (
        <form onSubmit={handleSubmit}>
            <div>
                <label htmlFor="name">Name:</label>
                <input 
                    type="text" 
                    id="name" 
                    name="name" 
                    value={form.name} 
                    onChange={handlerChangeForm} 
                />
            </div>
            <div>
                <label htmlFor="email">Email:</label>
                <input 
                    type="email" 
                    id="email" 
                    name="email" 
                    value={form.email} 
                    onChange={handlerChangeForm} 
                />
            </div>
            <button type="submit">Submit</button>
        </form>
    );
}

export default UserForm;
```
please refactor de code following this approach

File: src/components/Navbar/index.tsx [hunk 2]
```diff
@@ -1,20 +1,124 @@
...
+                <form className={`${styles.container} ${toogle ? styles.expanded : ''}`} onSubmit={handleSearch}>
+                    <input name='search' value={formData} onChange={handleChange} type="text" />
+                    <button type='submit' onClick={handleSearch}>buscar</button>
+                </form>
+                {
+                    toogle ? (
+                        <div className={styles.sectionCards}>
+                            {
+                                result.map((item, index) => (
+                                    <div key={index} className={styles.card}>
```
Comment (Ditmar) [inline src/components/Navbar/index.tsx:100]: Create small components. you can create another component with the name Cards or ItemCards, the best way to scale an application is by applying the separation of code.

**Review by Ditmar — COMMENTED** [review]
Good pr please read the comments an fix it.

**Review by Ditmar — APPROVED** [review]

**Comment by olmedo90** [comment]
image
It is according to the figma. continue with the same interest.
👍👍

**Pull request merged**